OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -25,6 +25,25 @@ protected Actor FindClosestEnemy(Squad owner)
 		{
 			return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition);
 		}
+
+		protected Actor GetRandomPreferredTarget(Squad owner)
+		{
+			var manager = owner.SquadManager;
+			var preferredTargets = manager.World.ActorsHavingTrait<MustBeDestroyed>(t => t.Info.RequiredForShortGame)
+					.Where(a => manager.IsPreferredEnemyUnit(a) && manager.IsNotHiddenUnit(a)).ToArray();
+
+			if (!preferredTargets.Any())
+				return FindClosestEnemy(owner);
+
+			return preferredTargets.Random(owner.World.LocalRandom);
+		}
+
+		protected Actor ScanThreat(Squad owner, Actor teamLeader, WDist scanRadius)
```

Comment:
This still has the same problems that were commented earlier - `ScanThreat` doesn't actually match what this is doing.

On closer inspection, this method is actually a direct copy of `SquadManager.FindClosestEnemy(WPos, WDist)`, so we should be using that instead of duplicating code.

Please either:
1. Change this to 
   ```csharp
   protected Actor FindClosestEnemy(Squad owner, WDist scanRadius)
   {
	   return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition, scanRadius);
   }
   ```
   and move the definition above `GetRandomPreferredTarget` so it is grouped with the other `FindClosestEnemy`.
2. Delete this and the `FindClosestEnemy` above completely, and have the 3 places that use them call `SquadManager.FindClosestEnemy` directly (IMO this is probably the better option).

Reply:
emmmm....
If you insist.

Reply:
2. is indeed a better Idea

Reply:
but if we follow the same principle as 2., we should also delete this:

```
		protected Actor FindClosestEnemy(Squad owner)
		{
			return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition);
		}
```